packages/bitcoin-js-wallet-provider/lib/BitcoinJsWalletProvider.ts

```diff
@@ -167,20 +119,77 @@ export default class BitcoinJsWalletProvider extends BitcoinWalletProvider(
       psbt.addInput(psbtInput)
     }
 
-    for (const output of targets) {
+    const outputs = targets
+    for (let i = 0; i < outputs.length; i++) {
+      outputs[i].address = addrToBitcoinJS(outputs[i].address, network)
+    }
+
+    for (const output of outputs) {
       psbt.addOutput(output)
     }
 
+    if (network.useBitcoinJSSign) {
```

Comment:
Would be great if these 2 methods of signing the psbt could be split into 2 functions